addon/session-stores/cookie.js

```diff
@@ -99,28 +98,34 @@ export default BaseStore.extend({
 
     @method persist
     @param {Object} data The data to persist
+    @return {Ember.RSVP.Promise} A promise that resolves when the data has been persisted and rejects otherwise.
     @public
   */
   persist(data) {
-    data           = JSON.stringify(data || {});
-    let expiration = this._calculateExpirationTime();
-    this._write(data, expiration);
-    this._lastData = this.restore();
+    return new RSVP.Promise((resolve) => {
+      data           = JSON.stringify(data || {});
+      let expiration = this._calculateExpirationTime();
+      this._write(data, expiration);
+      this.restore().then((restoredContent) => {
+        this._lastData = restoredContent;
+        resolve();
+      }, resolve);
```

Comment:
Resolving just hides the error in this case. I think the best option here is to just not use `restore` to set `_lastData` but simply set it directly so that there would be only one async operation going on here. This of course potentially requires other methods to be updated as well.